plugin/forward/persistent.go

```diff
@@ -96,15 +100,32 @@ Wait:
 
 			t.conns["tcp-tls"] = append(t.conns["tcp-tls"], &persistConn{conn, time.Now()})
 
+		case <-t.signal:
```

Comment:
consider reusing `stop` channel for both setting flag and actual closing connManager. The code which read from channel can distinguish between **read** and **close** operations by second boolean parameter, i.e.
```
case _, ok <-t.stop:
    if ok {
        // read from channel
        t.stopping = true
        continue
    }
    // channel was closed
    // close connection manager
    return
```